packages/react-components/react-components/stories/Concepts/Positioning/PositioningListenToUpdates.stories.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import {
+  useId,
+  Text,
+  makeStyles,
+  shorthands,
+  tokens,
+  Popover,
+  Button,
+  PopoverTrigger,
+  PopoverSurface,
+  PositioningImperativeRef,
+  PopoverProps,
+} from '@fluentui/react-components';
+
+const useStyles = makeStyles({
+  root: {
+    display: 'flex',
+    ...shorthands.gap('20px'),
+  },
+
+  button: {
+    display: 'block',
+    minWidth: '120px',
+  },
+
+  logContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+
+  logLabel: {
+    color: tokens.colorNeutralForegroundOnBrand,
+    backgroundColor: tokens.colorBrandBackground,
+    width: 'fit-content',
+    fontWeight: tokens.fontWeightBold,
+    ...shorthands.padding('2px', '12px'),
+  },
+
+  log: {
+    overflowY: 'auto',
+    boxShadow: tokens.shadow16,
+    position: 'relative',
+    minWidth: '200px',
+    height: '200px',
+    ...shorthands.border('2px', 'solid', tokens.colorBrandBackground),
+    ...shorthands.padding('12px', '12px'),
+  },
+});
+
+export const ListenToUpdates = () => {
+  const styles = useStyles();
+  const labelId = useId();
+  const [statusLog, setStatusLog] = React.useState<number[]>([]);
+  const positioningRef = React.useRef<PositioningImperativeRef>(null);
+  const [open, setOpen] = React.useState(false);
+
+  const onOpenChange: PopoverProps['onOpenChange'] = React.useCallback((e, data) => {
+    setOpen(data.open);
+    if (!data.open) {
+      setStatusLog([]);
+    }
+  }, []);
+
+  const updatePosition = React.useCallback(() => {
+    positioningRef.current?.updatePosition();
+  }, []);
+
+  const onPositioningEnd = React.useCallback(() => {
+    setStatusLog(s => [Date.now(), ...s]);
+  }, []);
+
+  return (
+    <div className={styles.root}>
+      <div>
+        <Popover
+          open={open}
+          onOpenChange={onOpenChange}
+          positioning={{ positioningRef, onPositioningEnd, position: 'below' }}
+        >
+          <PopoverTrigger>
+            <Button className={styles.button}>Open popover</Button>
+          </PopoverTrigger>
+          <PopoverSurface>
+            <Button className={styles.button} onClick={updatePosition}>
+              Update position
+            </Button>
+          </PopoverSurface>
+        </Popover>
+      </div>
+      <div className={styles.logContainer}>
+        <div className={styles.logLabel} id={labelId}>
+          Status log
+        </div>
+        <div role="log" aria-labelledby={labelId} className={styles.log}>
+          {statusLog.map((time, i) => {
+            const date = new Date(time);
+            return (
+              <div key={i}>
+                {date.toLocaleTimeString()} <Text weight="bold">Position updated</Text>
```

Comment:
```suggestion
                {date.toLocaleTimeString()} <Text weight="bold">Position updated [{i}]</Text>
```

nit: let's add `i` to output, otherwise it's unclear what happens on scroll:

<img width="272" alt="image" src="https://github.com/microsoft/fluentui/assets/14183168/32c5bfd0-015d-4f48-8d73-e0661ce82f58">